Add test scenario specification for Zone11

marc-vanderwal opened the pull request:
## Purpose

This PR adds a test scenario specification for Zone11 that matches the test case specification as worded in #1415.

Of note here are the scenarios that perform tests on three different root zones. They will require three changes to the root hints during unit testing.

## Context

See #1256 (issue about Zone11).

Depends on #1415 (Zone11 test case specification update).

## Changes

- Add test scenario specification for Zone11.

## How to test this PR

Review.

marc-vanderwal committed:
Add test scenario specification for Zone11

The overhaul of the test specification for Zone11 (see #1415) is a good
opportunity to add some test scenarios for that same test case.

Of note here are the scenarios that perform tests on three different
root zones. They will require three changes to the root hints during
unit testing.

marc-vanderwal committed:
Zone11: add test zones and CoreDNS configuration

Add the test zones and the CoreDNS configuration files implementing the
test scenario specification for Zone11.

matsduf commented on docs/public/specifications/test-zones/Zone-TP/zone11.md:
```diff
@@ -0,0 +1,325 @@
+
+* The child zone is `SCENARIO.zone11.xa`.
+  * There is a zone file for the child zone.
+  * It is served by three name servers named ns1, ns2 and ns3.child.zone11.xa.
```
For the default scenarios I include one or two NS to make configuration smaller. What do you think will be gained by three?

marc-vanderwal replied:
There are two scenarios where I really need three: SPF-MISSING-ON-ONE and ALL-DIFFERENT-SPF. For all other scenarios, two are enough.

matsduf reviewed (dismissed):
The test zones are not included.

marc-vanderwal commented:
> The test zones are not included.

Yes, they’re coming, but I wanted some early feedback on the outline of the test scenarios before completing them.

marc-vanderwal commented on docs/public/specifications/test-zones/Zone-TP/zone11.md:
```diff
@@ -0,0 +1,325 @@
+the apex.
+
+* Zone: same-spf-different-txt.zone11.xa
+  * ns1 serves a TXT RRset containing one SPF record and one non-SPF record.
+  * ns2 and ns3 serve a different TXT RRset containing the same SPF record as
+    ns1.
```
Yes, that’s right. I’ll clarify.

matsduf commented on test-zone-data/Zone-TP/zone11/zone11.xa.zone:
```diff
@@ -0,0 +1,78 @@
+ns1b.invalid-syntax-2   A     127.19.11.31
+                        AAAA  fda1:b2:c3::127:19:11:31
+ns2.invalid-syntax-2    A     127.19.11.32
+                        AAAA  fda1:b2:c3::127:19:11:32
```
It looks like the text should read "**ns1a**.invalid-syntax-2 deliberately has the same IPv4 and IPv6 addresses as **ns1b**.invalid-syntax-2"

marc-vanderwal committed:
Zone11: add output for test scenarios

Add the output for test scenarios.

marc-vanderwal committed:
Fix delegation inconsistency in test arpa. zone

The parent (root zone) delegates arpa. to ns1. and ns2., but the NS
records at the apex of the arpa. zone listed different name servers:
ns1.xa and ns2.xa. This inconsistency caused some odd errors when trying
to run zonemaster-cli on those test zones, while preparing the
test-zones-output.md.

This might be the sign of a bug somewhere in Basic02, which should
warrant further investigation.

matsduf commented on test-zone-data/COMMON/arpa:
```diff
@@ -45,6 +45,10 @@ $ORIGIN 22.9.19.127.in-addr.arpa.
+$ORIGIN zone11.arpa.
+@                       NS   ns1.zone11.xa.
+@                       NS   ns2.zone11.xa.
+
```
This is one way to do it. I have moved to creating test case dedicated root, xa and other zones instead to have as few shared files as possible to cut down the risk of conflicts

matsduf reviewed (commented):
Looks fine. The final test is when used as unit tests.

tgreenx commented on docs/public/specifications/test-zones/Zone-TP/zone11.md:
```diff
@@ -0,0 +1,340 @@
+A root zone without SPF records.
+
+* Zone: "." (root zone)
+  * It is served by ns-no-spf.root-servers.zone11.xa.
```
```suggestion
  * It is served by ns1 and ns2.no-spf.root-servers.zone11.xa.
```

tgreenx commented on docs/public/specifications/test-zones/Zone-TP/zone11.md:
```diff
@@ -0,0 +1,340 @@
+
+### SERVFAIL
+
+A zone where queries for the TXT RRset are responded to with an RCODE
```
```suggestion
A zone where queries for the TXT RRset are responded with an RCODE
```

marc-vanderwal replied:
It’s actually grammatically correct, but I’ll rephrase anyway.

tgreenx commented on test-zone-data/Zone-TP/zone11/zone11.xa.zone:
```diff
@@ -0,0 +1,80 @@
+ns3.child               A     127.19.11.33
+                        AAAA  fda1:b2:c3::127:19:11:33
+
+;; TODO: do the same for all other scenarios in zone11.cfg
```
Can be removed?
```suggestion
```

tgreenx reviewed (commented):
A few minor comments otherwise LGTM

matsduf reviewed (commented):
Looks fine.

matsduf commented:
@marc-vanderwal, please merge.
